tools/sdk-generation-pipeline/documents/docker/docker-image-design.md

```diff
@@ -0,0 +1,25 @@
+# ARCHITECTURE DIAGRAM
+![docker design](docker-image-design.md)
+
+The core of the docker image is task engine, which contains four tasks: Init Task, Generate and Build Task, Mock Test Task. There is a configuration file in each sdk repository, and it defines which task should be executed. To serve different users/pipeline, we provide different docker commands.  Also, after the tasks are executed, there are some outputs, such as generated codes, task execution result, which can be used by following steps in pipeline or service team.
+
+# TASK ENGINE
+There are mainly four tasks defined in task engine: Init Task, Generate and Build Task, Mock Test Task, Live Test Task. Task engine executes these tasks based on a configuration file in sdk repository and you can find [the schema of configuration file here](../task-engine/schema/CodegenToSdkConfigSchema.json).
```

Comment:
We should also add a document with examples of the configuration file instead of just pointing a full schema.

Reply:
add a link to the example doc